### PR TITLE
Deploy app from Azure Storage account artifact

### PR DESCRIPTION
**Pull request by ryanvog**

### CODE REVIEW

File: Artifacts/linux-deploy-app-from-storage/Artifactfile.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "destination": {
+      "type": "string",
+      "displayName": "Destination",
+      "description": "The destination folder on the VM to deploy the app.",
+      "defaultValue": "/opt/<myapp>"
+    },
+    "storageConnectionString": {
+      "type": "string",
+      "displayName": "Storage Connection String",
+      "description": "the connection string to the Azure storage account container containing the app."
```
Comment (drwill-ms) [inline Artifacts/linux-deploy-app-from-storage/Artifactfile.json]: capital T in the

File: Artifacts/linux-deploy-app-from-storage/Artifactfile.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    },
+    "deployScript": {
+      "type": "string",
+      "displayName": "Deploy script file",
```
Comment (drwill-ms) [inline Artifacts/linux-deploy-app-from-storage/Artifactfile.json]: Title Casing

File: Artifacts/linux-deploy-app-from-storage/BlobDownloadService.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                    filePath,
+                    function (err, serverBlob) {
+                        blobsDownloaded += 1;
+                        self._blobs.push(serverBlob);
+
+                        if (err) {
+                            callback(err, null);
+                        }
+                        else if ((result.continuationToken === null) && (blobsDownloaded >= blobCount)) {
+                            callback(null, self._blobs);
```
Comment (drwill-ms) [inline Artifacts/linux-deploy-app-from-storage/BlobDownloadService.js]: add code comment: done fetching pages?

File: Artifacts/linux-deploy-app-from-storage/download_azure_container.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    container: process.argv[2],
+    destination: process.argv[3],
+    connectionString: process.argv[4]
+};
+
+// catch the uncaught errors that weren't wrapped in a domain or try catch statement
+// do not use this in modules, but only in applications, as otherwise we could have multiple of these bound
+process.on('uncaughtException', function (err) {
+    console.error(err);
+    process.exit(127);
```
Comment (drwill-ms) [inline Artifacts/linux-deploy-app-from-storage/download_azure_container.js]: What is 127?

**Comment by drwill-ms** [comment]
LGTM

File: Artifacts/linux-deploy-app-from-storage/Artifactfile.json (hunk 1 above)
Comment (ryanvog) [inline Artifacts/linux-deploy-app-from-storage/Artifactfile.json]: fixed.

File: Artifacts/linux-deploy-app-from-storage/Artifactfile.json (hunk 2 above)
Comment (ryanvog) [inline Artifacts/linux-deploy-app-from-storage/Artifactfile.json]: fixed.

File: Artifacts/linux-deploy-app-from-storage/BlobDownloadService.js (hunk 3 above)
Comment (ryanvog) [inline Artifacts/linux-deploy-app-from-storage/BlobDownloadService.js]: fixed.

File: Artifacts/linux-deploy-app-from-storage/download_azure_container.js (hunk 4 above)
Comment (ryanvog) [inline Artifacts/linux-deploy-app-from-storage/download_azure_container.js]: Switched to a more common error exit code of 1

**Pull request merged**